apps/web/src/services/asset-inventory/components/ProviderList.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+    <div class="provider-list"
+         :class="{'low-count-provider-list': providerList?.length < 6}"
+    >
+        <p-button v-for="item in providerList"
+                  :key="item.key"
+                  :class="{'selected-button' : item.key === selectedProvider}"
+                  style-type="tertiary"
+                  @click="handleSelectProvider(item.key)"
+        >
+            <p-lazy-img v-if="item.icon"
+                        :src="item.icon"
+                        width="1rem"
+                        height="1rem"
+            />
+            <p class="provider-name">
+                {{ item.name }}
+            </p>
+        </p-button>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { reactive } from 'vue';
+
+import { PLazyImg, PButton } from '@spaceone/design-system';
+
+import type { ReferenceItem } from '@/store/modules/reference/type';
+
+import { useProxyValue } from '@/common/composables/proxy-state';
+
+interface Props {
+    providerList?: ReferenceItem[];
+    selectedProvider: string;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+    providerList: undefined,
+    selectedProvider: 'all',
+});
+
+const emit = defineEmits<{(e: 'handle-selected-provider', providerName: string): void}>();
+
+const state = reactive({
+    proxySelectedProvider: useProxyValue('selectedProvider', props, emit),
+});
+const handleSelectProvider = (providerName) => {
+    state.proxySelectedProvider = providerName;
+    emit('handle-selected-provider', providerName);
```

Comment:
suggestion: When emitting, it's better to use intuitive names rather than using the event name as it is. Like this-

```suggestion
    emit('change-provider', providerName);
```